src/components/display/Table/types/BasicTable/index.tsx

```diff
@@ -0,0 +1,219 @@
+import React, {useCallback, useRef, useState} from "react";
+import {useSortBy, useTable} from "react-table";
+
+// Styles
+import {
+    ColumnsSelector,
+    HeaderRow, IconContainer,
+    Layout,
+    OptionItem,
+    OptionList, SortIcon,
+    TableView,
+    TextContainer,
+    ThView,
+    View
+} from "../../styles";
+import {dark} from "../../../../../theme";
+
+// Components
+import Typography from "../../../Typography";
+import CheckBox from "../../../../inputs/CheckBox";
+import IconButton from "../../../../inputs/IconButton";
+
+// Assets
+import OptionsDotsIcon from "../../../../../assets/icons/option-dots-icon.svg";
+
+// Types
+import {TableBasicProps} from "../../types";
+export default function BasicTable({
+    columns = [],
+    data = [],
+    onColumnHidden = () => {},
+    defaultHiddenColumns,
+    hideOptionsButton,
+    isUserTable,
+    initialState = {},
+    hasFooter = false
+  }: TableBasicProps) {
+    // Refs
+    const tableRef = useRef(null);
+
+    // States
+    const [hiddenColumns, setHiddenColumns] = useState<string[]>(defaultHiddenColumns || []);
+
+    // Hooks
+    const { getTableProps, getTableBodyProps, rows, headerGroups, footerGroups, toggleHideColumn, prepareRow } = useTable({
+            columns,
+            data,
+            initialState
+        },
+        useSortBy
+    );
+
+    const firstPageRows = rows.slice(0, 20);
```

Comment:
Can we make pagination here customizable? If i want more then 20 slices for example